scripts/windows/setup/Install-SecurityDaemon.ps1

```diff
@@ -150,12 +174,29 @@ function Get-SecurityDaemon {
     }
 }
 
-function Set-Path {
-    $SystemEnvironment = "HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment"
+function Remove-SecurityDaemonResources {
     $EdgePath = "C:\ProgramData\iotedge"
-    $SystemPath = (Get-ItemProperty -Path $SystemEnvironment -Name Path).Path
-    if ($SystemPath -notlike "*$EdgePath*") {
-        Set-ItemProperty -Path $SystemEnvironment -Name Path -Value "$SystemPath;$EdgePath"
+    Remove-Item -Recurse $EdgePath -ErrorAction SilentlyContinue -ErrorVariable CmdErr
```

Comment:
Can switch the order of deletion here? First remove registry entries and then remove the binaries. Sometimes I have seen binaries removal fail because they are open elsewhere, haven't been able to root cause exactly, but have had better luck if the registry entries are deleted first.

Reply:
Sure no problem. Although I don't see how deleting the registry entry first increases the chances that folder deletion will succeed?

Reply:
Done